esp32/src/sensors.cpp

```diff
@@ -70,157 +76,127 @@ void collectIMUData(DataPacket &packet) {
-            DEBUG_SERIAL.print(", ");
-            DEBUG_SERIAL.print(sensorValue.un.rotationVector.j);
-            DEBUG_SERIAL.print(", ");
-            DEBUG_SERIAL.print(sensorValue.un.rotationVector.k);
-            DEBUG_SERIAL.print(", ");
-            DEBUG_SERIAL.println(sensorValue.un.rotationVector.real);
-            #endif
-            float norm = sqrt(
-              sensorValue.un.rotationVector.i * sensorValue.un.rotationVector.i +
-              sensorValue.un.rotationVector.j * sensorValue.un.rotationVector.j +
-              sensorValue.un.rotationVector.k * sensorValue.un.rotationVector.k +
-              sensorValue.un.rotationVector.real * sensorValue.un.rotationVector.real
-            );
-            if (norm > 0) {
-              packet.quat_x = sensorValue.un.rotationVector.i / norm;
-              packet.quat_y = sensorValue.un.rotationVector.j / norm;
-              packet.quat_z = sensorValue.un.rotationVector.k / norm;
-              packet.quat_w = sensorValue.un.rotationVector.real / norm;
-            } else {
-              packet.quat_x = 0.0f;
-              packet.quat_y = 0.0f;
-              packet.quat_z = 0.0f;
-              packet.quat_w = 1.0f;
-            }
-          } else {
-            packet.quat_x = sensorValue.un.rotationVector.i;
-            packet.quat_y = sensorValue.un.rotationVector.j;
-            packet.quat_z = sensorValue.un.rotationVector.k;
-            packet.quat_w = sensorValue.un.rotationVector.real;
-          }
-          executed_cases |= STATUS_BNO08X_ROT;
-          break;
-
-        case SH2_MAGNETIC_FIELD_CALIBRATED:
-          if (abs(sensorValue.un.magneticField.x) > MAX_MAG_VALUE ||
-              abs(sensorValue.un.magneticField.y) > MAX_MAG_VALUE ||
-              abs(sensorValue.un.magneticField.z) > MAX_MAG_VALUE) {
-            badIMUDataDetected = true;
-            #if DEBUG_MODE
-            DEBUG_SERIAL.print("Bad mag data: ");
-            DEBUG_SERIAL.print(sensorValue.un.magneticField.x);
-            DEBUG_SERIAL.print(", ");
-            DEBUG_SERIAL.print(sensorValue.un.magneticField.y);
-            DEBUG_SERIAL.print(", ");
-            DEBUG_SERIAL.println(sensorValue.un.magneticField.z);
-            #endif
-            packet.magnetic_x = constrain(sensorValue.un.magneticField.x, -MAX_MAG_VALUE, MAX_MAG_VALUE);
-            packet.magnetic_y = constrain(sensorValue.un.magneticField.y, -MAX_MAG_VALUE, MAX_MAG_VALUE);
-            packet.magnetic_z = constrain(sensorValue.un.magneticField.z, -MAX_MAG_VALUE, MAX_MAG_VALUE);
-          } else {
-            packet.magnetic_x = sensorValue.un.magneticField.x;
-            packet.magnetic_y = sensorValue.un.magneticField.y;
-            packet.magnetic_z = sensorValue.un.magneticField.z;
-          }
-          executed_cases |= STATUS_BNO08X_MAG;
-          break;
-      }
-    } else {
-      attemptReset = true;
-    }
+  if (bno08x.wasReset()) {
+    Serial.print("BNO08x was reset ");
```

Comment:
```suggestion
    DEBUG_SERIAL.print("BNO08x was reset ");
```
important to not bug out when reading serial from python